apps/transport/test/transport_web/controllers/discussion_controller_test.exs

```diff
@@ -0,0 +1,65 @@
+defmodule TransportWeb.DiscussionControllerTest do
+  use TransportWeb.ConnCase, async: true
+  import DB.Factory
+  import Mox
+
+  setup :verify_on_exit!
+
+  setup do
+    Ecto.Adapters.SQL.Sandbox.checkout(DB.Repo)
+  end
+
+  describe "post_discussion" do
+    test "success case", %{conn: conn} do
+      %DB.Dataset{datagouv_id: datagouv_id} = dataset = insert(:dataset)
+      title = "J'ai une question"
+      comment = "Coucou"
+
+      Datagouvfr.Client.Discussions.Mock
+      |> expect(:post, fn %Plug.Conn{}, ^datagouv_id, ^title, ^comment ->
+        {:ok, nil}
+      end)
+
+      conn =
+        conn
+        |> Phoenix.ConnTest.init_test_session(%{current_user: %{"is_admin" => false}})
+        |> post(
+          discussion_path(conn, :post_discussion, datagouv_id, %{
+            "comment" => comment,
+            "title" => title,
+            "dataset_slug" => dataset.slug
+          })
+        )
+
+      assert redirected_to(conn, 302) == dataset_path(conn, :details, dataset.slug)
+      assert Phoenix.Flash.get(conn.assigns.flash, :info) =~ "Nouvelle discussion commencée"
+    end
+  end
+
+  describe "post_answer" do
+    test "success case", %{conn: conn} do
+      dataset = insert(:dataset)
+      discussion_id = Ecto.UUID.generate()
+      comment = "Coucou"
+
+      Datagouvfr.Client.Discussions.Mock
+      |> expect(:post, fn %Plug.Conn{}, ^discussion_id, ^comment, close: true ->
+        {:ok, nil}
+      end)
+
+      conn =
+        conn
+        |> Phoenix.ConnTest.init_test_session(%{current_user: %{"is_admin" => false}})
+        |> post(
+          discussion_path(conn, :post_answer, dataset.datagouv_id, discussion_id, %{
+            "comment" => comment,
+            "dataset_slug" => dataset.slug,
+            "answer_and_close" => true
+          })
+        )
+
+      assert redirected_to(conn, 302) == dataset_path(conn, :details, dataset.slug)
+      assert Phoenix.Flash.get(conn.assigns.flash, :info) =~ "Réponse publiée"
+    end
```

Comment:
Peut-être rajouter un test dans le cas où l’API datagouv répond «nope», par exemple si la personne n’a pas le droit de clore la discussion ? Voir ici https://doc.data.gouv.fr/api/reference/#/discussions/comment_discussion ils renvoient une 403 pour : "Not allowed to close this discussion OR can't add comments on a closed discussion"